src/udev/udevd.c

```diff
@@ -1544,59 +1554,55 @@ static int on_sighup(sd_event_source *s, const struct signalfd_siginfo *si, void
         return 1;
 }
 
-static int on_sigchld(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
-        Manager *manager = userdata;
+static int on_sigchld(sd_event_source *s, const siginfo_t *si, void *userdata) {
+        Manager *manager = ASSERT_PTR(userdata);
```

Comment:
as mentioned above: if `userdata` would be the worker object, rather than the manager object, you wouldn't have to look up the worker via the hashmap by the pid.

and the worker has a `manager` field anyway, where you can acquire the manager object trivial if you need it